src/disk.hpp

```diff
@@ -217,45 +217,57 @@ struct FileDisk : Disk {
 
 struct BufferedDisk : Disk
 {
-    BufferedDisk(Disk* disk, uint64_t file_size) : disk_(disk), file_size_(file_size) {}
+    BufferedDisk(FileDisk* disk, uint64_t file_size) : disk_(disk), file_size_(file_size) {}
 
-    void Read(uint64_t begin, uint8_t *memcache, uint64_t length) override
+    uint8_t const* Read(uint64_t begin, uint64_t length) override
```

Comment:
What if the requested read is bigger than `uint64_t read_ahead` ?
Since we pass a pointer out, a valid read can only be as large as that buffer.

Reply:
we only read one entry at a time, and I think the largest entry is 7 bytes. I suppose there could be an assert though